job.go

```diff
@@ -189,182 +105,13 @@ func (j *Job) periodDuration() (time.Duration, error) {
 }
 
-// roundToMidnight truncate time to midnight
-func (j *Job) roundToMidnight(t time.Time) time.Time {
-	return time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, j.loc)
-}
-
-// scheduleNextRun Compute the instant when this job should run next
-func (j *Job) scheduleNextRun() error {
-	now := time.Now()
-	if j.lastRun == time.Unix(0, 0) {
-		j.lastRun = now
-	}
-
-	if j.nextRun.After(now) {
-		return nil
-	}
-
-	periodDuration, err := j.periodDuration()
-	if err != nil {
-		return err
-	}
-
-	switch j.unit {
-	case seconds, minutes, hours:
-		j.nextRun = j.lastRun.Add(periodDuration)
-	case days:
-		j.nextRun = j.roundToMidnight(j.lastRun)
-		j.nextRun = j.nextRun.Add(j.atTime)
-	case weeks:
-		j.nextRun = j.roundToMidnight(j.lastRun)
-		dayDiff := int(j.startDay)
-		dayDiff -= int(j.nextRun.Weekday())
-		if dayDiff != 0 {
-			j.nextRun = j.nextRun.Add(time.Duration(dayDiff) * 24 * time.Hour)
-		}
-		j.nextRun = j.nextRun.Add(j.atTime)
-	}
-
-	// advance to next possible schedule
-	for j.nextRun.Before(now) || j.nextRun.Before(j.lastRun) {
-		j.nextRun = j.nextRun.Add(periodDuration)
-	}
-
-	return nil
-}
-
 // NextScheduledTime returns the time of when this job is to run next
 func (j *Job) NextScheduledTime() time.Time {
 	return j.nextRun
 }
 
-// set the job's unit with seconds,minutes,hours...
-func (j *Job) mustInterval(i uint64) error {
-	if j.interval != i {
-		return fmt.Errorf("interval must be %d", i)
-	}
-	return nil
-}
-
-// From schedules the next run of the job
-func (j *Job) From(t *time.Time) *Job {
-	j.nextRun = *t
-	return j
-}
-
-// setUnit sets unit type
-func (j *Job) setUnit(unit timeUnit) *Job {
-	j.unit = unit
-	return j
-}
-
-// Seconds set the unit with seconds
-func (j *Job) Seconds() *Job {
-	return j.setUnit(seconds)
-}
-
-// Minutes set the unit with minute
-func (j *Job) Minutes() *Job {
-	return j.setUnit(minutes)
-}
-
-// Hours set the unit with hours
-func (j *Job) Hours() *Job {
-	return j.setUnit(hours)
-}
-
-// Days set the job's unit with days
-func (j *Job) Days() *Job {
-	return j.setUnit(days)
-}
-
-// Weeks sets the units as weeks
-func (j *Job) Weeks() *Job {
-	return j.setUnit(weeks)
-}
-
-// Second sets the unit with second
-func (j *Job) Second() *Job {
-	j.mustInterval(1)
-	return j.Seconds()
-}
-
-// Minute sets the unit  with minute, which interval is 1
-func (j *Job) Minute() *Job {
-	j.mustInterval(1)
-	return j.Minutes()
-}
-
-// Hour sets the unit with hour, which interval is 1
-func (j *Job) Hour() *Job {
-	j.mustInterval(1)
-	return j.Hours()
-}
-
-// Day sets the job's unit with day, which interval is 1
-func (j *Job) Day() *Job {
-	j.mustInterval(1)
-	return j.Days()
-}
-
-// Week sets the job's unit with week, which interval is 1
-func (j *Job) Week() *Job {
-	j.mustInterval(1)
-	return j.Weeks()
-}
-
-// Weekday start job on specific Weekday
-func (j *Job) Weekday(startDay time.Weekday) *Job {
-	j.mustInterval(1)
-	j.startDay = startDay
-	return j.Weeks()
-}
-
```

Comment:
Do we use the `GetWeekday` func anywhere? Or should that be removed ⬇️

Reply:
I believe we could leave it for now, it may be useful to get information from a job once it's created